gui/src/app/view/coins.rs

```diff
@@ -74,36 +74,43 @@ fn coin_list_view(
                                     )
                                 } else {
-                                    if let Some(b) = coin.block_height {
-                                        if blockheight > b as u32 + timelock {
-                                            Some(Container::new(
-                                                Row::new()
-                                                    .spacing(5)
-                                                    .push(text(" 0").small().style(color::ALERT))
-                                                    .push(
-                                                        icon::hourglass_done_icon()
-                                                            .small()
-                                                            .style(color::ALERT),
-                                                    )
-                                                    .align_items(Alignment::Center),
-                                            ))
-                                        } else {
-                                            Some(Container::new(
-                                                Row::new()
-                                                    .spacing(5)
-                                                    .push(
-                                                        text(format!(
-                                                            " {}",
-                                                            b as u32 + timelock - blockheight
-                                                        ))
-                                                        .small(),
-                                                    )
-                                                    .push(icon::hourglass_icon().small())
-                                                    .align_items(Alignment::Center),
-                                            ))
-                                        }
+                                    let seq = remaining_sequence(coin, blockheight, timelock);
+                                    if seq == 0 {
+                                        Some(Container::new(
+                                            Row::new()
+                                                .spacing(5)
+                                                .push(text(" 0").small().style(color::ALERT))
+                                                .push(
+                                                    icon::hourglass_done_icon()
+                                                        .small()
+                                                        .style(color::ALERT),
+                                                )
+                                                .align_items(Alignment::Center),
+                                        ))
+                                    } else if seq < timelock * 10 / 100 {
```

Comment:
Would be nice to merge this and the above computation so that only one place needs to be updated should a contributor want to modify this in the future?